### PR TITLE
Clarify data representation of enums.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/4624

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -355,6 +355,10 @@ Enums are one way to create a user-defined type in Solidity. They are explicitly
 to and from all integer types but implicit conversion is not allowed.  The explicit conversions
 check the value ranges at runtime and a failure causes an exception.  Enums needs at least one member.
 
+The data representation is the same as for enums in C: The options are represented by a sequence
```
Comment (axic) [inline docs/types.rst]: Strict sequence?

Comment (chriseth) [reply docs/types.rst]: What is a strict sequence? Perhaps "sequence of subsequent integers"? Any suggestions welcome...

Comment (axic) [reply docs/types.rst]: "Increasing sequence without gaps?"

Comment (mattaereal) [reply docs/types.rst]: _a subsequent list of natural numbers_

Comment (chriseth) [reply docs/types.rst]: Thanks! I thought about 'subsequent sequence' first, but that didn't really cut it :)

**Comment by chriseth** [comment]
What about this version?

File: docs/types.rst [hunk 2]
```diff
@@ -384,7 +388,7 @@ check the value ranges at runtime and a failure causes an exception.  Enums need
         // Since enum types are not part of the ABI, the signature of "getChoice"
         // will automatically be changed to "getChoice() returns (uint8)"
         // for all matters external to Solidity. The integer type used is just
-        // large enough to hold all enum values, i.e. if you have more values,
+        // large enough to hold all enum values, i.e. if you have more than 255 values,
```
Comment (leonardoalt) [inline docs/types.rst]: 256 values should still be fine for `uint8`, right? 0 to 255

Comment (chriseth) [reply docs/types.rst]: of course!

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**